website/pages/api-docs/secret/key-management/index.mdx

```diff
@@ -0,0 +1,540 @@
+
+- `provider` `(string: <required>)` – Specifies the name of a KMS provider that's external to Vault.
+  Cannot be changed after creation. The following values are supported:
+  - `azurekeyvault` - KMS provider represents a named
+  [Azure Key Vault](https://docs.microsoft.com/en-us/azure/key-vault/) instance.
+
+- `key_collection` `(string: <required>)` – Refers to a location to store keys in the specified
+  provider. Cannot be changed after creation. The expected value for this parameter will differ
+  depending on the specified provider. The following values are expected for each provider:
+  - `azurekeyvault` - The name of an existing Azure Key Vault instance.
+
+- `credentials` `(map<string|string>: nil)` – The credentials to use for authentication with
+  the specified provider. Supplying values for this parameter is optional, as credentials may
+  also be specified as environment variables. Environment variables will take precedence over
+  credentials provided via this parameter. The expected keys and values for this parameter
+  will differ depending on the specified provider.
+
+  The following keys and values are expected for each provider:
+    - `azurekeyvault`
+      - `tenant_id` `(string: <required>)` - The tenant ID for the Azure Active Directory
+      organization. May also be specified by the `AZURE_TENANT_ID` environment variable.
+      - `client_id` `(string: <required or MSI>)` - The client ID for credentials to invoke the
+      Azure APIs. May also be specified by the `AZURE_CLIENT_ID` environment variable.
+      - `client_secret` `(string: <required or MSI>)` - The client secret for credentials to invoke
+      the Azure APIs. May also be specified by the `AZURE_CLIENT_SECRET` environment variable.
+      - `environment` `(string: "AzurePublicCloud")` - The Azure Cloud environment API endpoints to
+      use. May also be specified by the `AZURE_ENVIRONMENT` environment variable.
+
+### Sample Payload
+
+```json
+{
+  "credentials": [
+    "client_id=example-client-id",
+    "client_secret=example-client-secret",
+    "tenant_id=example-tenant-id"
+  ],
+  "key_collection": "example-keyvault-name",
+  "provider": "azurekeyvault"
+}
+```
+
+### Sample Request
+
+```shell-session
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request PUT \
+    --data @payload.json \
+    http://127.0.0.1:8200/v1/keymgmt/kms/example-kms
+```
+
+## Read KMS Provider
+
+This endpoint returns information about a KMS provider.
+
+| Method | Path                  |
+| :----- | :-------------------- |
+| `GET`  | `/keymgmt/kms/:name`  |
+
+### Parameters
+
+- `name` `(string: <required>)` – Specifies the name of the KMS provider to read.
+  This is provided as part of the request URL.
+
+### Sample Request
+
+```shell-session
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request GET \
+    http://127.0.0.1:8200/v1/keymgmt/kms/example-kms
+```
+
+### Sample Response
+
+```json
+{
+  "data": {
```

Comment:
Does it make sense to return the `tenant_id` and `environment` here? Assuming it's possible to manage multiple Azure KMS with different values

Reply:
I think it would make sense to return those items here. Unfortunately, it would require a code change, which we're a little late for at this point. I think this would be a good item to gather feedback on in the tech preview.